charts/linkerd-control-plane/values.yaml

```diff
@@ -153,6 +153,11 @@ proxy:
   logLevel: warn,linkerd=info,trust_dns=error
   # -- Log format (`plain` or `json`) for the proxy
   logFormat: plain
+  # -- (`off` or `insecure`) If set to `off`, will prevent the proxy from
+  # logging HTTP headers. If set to `insecure`, HTTP headers may be logged
+  # verbatim. Note that setting this to `insecure` is not alone sufficient to
+  # log HTTP headers; the proxy logLevel must also be set to debug.
+  logHTTPHeaders: "off"
```

Comment:
I prefer having a regular boolean here like everywhere else instead of introducing special values. E.g. if people use "true" that would get misinterpreted in the template as "off" (according to the change in _proxy.tpl). To avoid that you'd have to validate the possible values, which we don't do for regular booleans as those are "standard" (true, >0, non-empty vs false, 0, empty).

Reply:
The motivation for not using a boolean here is that using a scary sounding value like "insecure" helps users realize that enabling this has security implications.  It also leaves the door open to adding other modes in the future like "hash", "redact", or "obfuscate".